crates/goose/src/session/session_manager.rs

```diff
@@ -44,6 +44,18 @@ pub struct Session {
     pub message_count: usize,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
+pub struct TokenState {
+    /// Current turn token counts
+    pub input_tokens: Option<i32>,
+    pub output_tokens: Option<i32>,
+    pub total_tokens: Option<i32>,
+    /// Accumulated token counts across all turns
+    pub accumulated_input_tokens: Option<i32>,
+    pub accumulated_output_tokens: Option<i32>,
+    pub accumulated_total_tokens: Option<i32>,
+}
```

Comment:
this seems like a good idea to have, but I would expect then to also have a way to read this on a session and a way to set this on the builder? would probably clean up the rest of the code